xbmc/utils/SystemInfo.h

```diff
@@ -10,6 +10,7 @@
 
 #include "InfoLoader.h"
 #include "settings/ISubSettings.h"
+#include "utils/Job.h"
```

Comment:
is forward  declaration of CJob sufficient?

Reply:
> class CSysInfoJob : public CJob

No, the include was missing before. The class derived from CJob already existed in this header.

Reply:
my point is the include should go into `SystemInfo.cpp`

```diff
diff --git a/xbmc/utils/SystemInfo.cpp b/xbmc/utils/SystemInfo.cpp
index bc0aed6765..d5f4c70bd8 100644
--- a/xbmc/utils/SystemInfo.cpp
+++ b/xbmc/utils/SystemInfo.cpp
@@ -26,6 +26,7 @@
 #include "settings/Settings.h"
 #include "settings/SettingsComponent.h"
 #include "utils/CPUInfo.h"
+#include "utils/Job.h"
 #include "utils/log.h"
 
 #ifdef TARGET_WINDOWS
diff --git a/xbmc/utils/SystemInfo.h b/xbmc/utils/SystemInfo.h
index 25ec438515..ccca8729bf 100644
--- a/xbmc/utils/SystemInfo.h
+++ b/xbmc/utils/SystemInfo.h
@@ -10,7 +10,6 @@
 
 #include "InfoLoader.h"
 #include "settings/ISubSettings.h"
-#include "utils/Job.h"
 
 #include <string>
 #include <vector>
@@ -24,6 +23,7 @@
 
 #define REG_CURRENT_VERSION L"Software\\Microsoft\\Windows NT\\CurrentVersion"
 
+class CJob;
 
 class CSysData
 {

```

Reply:
CSysInfoJob is declared in the header file. As it is derived from CJob, the bar class must be included - concept of self included header. The code may compile by luck, because CJob.h for included already from elsewhere.

Reply:
got you, thanks 👍

Reply:
And I was missing vector as well, fixed in the last push